tests/unit/auth/test_sigv4.py

```diff
@@ -507,6 +507,9 @@ def test_canonical_request(self):
 
     def test_non_string_headers(self):
         self.awesome_bucket_request.headers['Content-Length'] = 8
+        # Ensures headers are alphabetized, not 'header:value' strings:
```

Comment:
This comment is confusing--I thought you wanted "header:value" strings.

Reply:
You only want the header keys to be alphabetized -- the current behavior is a bug